lang/elaborator/src/typechecker/decls/definition.rs

```diff
@@ -33,7 +33,7 @@ impl CheckToplevel for Def {
 
             let ws = WithScrutinee { cases, scrutinee: self_param_nf.expect_typ_app()? };
             ws.check_exhaustiveness(ctx)?;
-            let cases = ws.check_ws(ctx, &ret_typ_nf)?;
+            let cases = ws.check_ws(ctx, &ret_typ_nf, span)?;
```

Comment:
Do we want the span of the definition or of the case here? The same applies to top-level codefinitions and local (co)matches.

Reply:
![image](https://github.com/user-attachments/assets/bf5e3838-e477-43d0-bbe4-e4d4295ef6f4)

Hmm; for a simple example, this is how it looks right now. I take it you would like it better if the `n` is highlighted where it is part of the scrutinee?

```
data Nat { Z, S(n: Nat) }

def Nat.add(m: Nat): Nat {
    Z => m,
    S(n) => S((n : Type).add(m))
    --^ i.e. here?
}
```

Reply:
No, I think the span in your example above is reasonable.
I'm talking about examples where unification fails for top-level (co)definitions and expression-level (co)matches.
For instance, consider the following example:

```
data Bool { T, F }

data Foo(a: Bool) {
    Bar: Foo(T),
    Baz: Foo(F)
}

def Foo(?).foo: Foo(F) {
    Bar => ?,
    Baz => ?
}
```

The corresponding error message is:

```
Error: T-016

  × Cannot automatically decide whether T and ? unify
    ╭─[file:///home/tim/Dev/polarity-lang/polarity/foo.pol:3:1]
  3 │     data Foo(a: Bool) {
  4 │ ╭─▶     Bar: Foo(T),
    · │             ─
  5 │         Baz: Foo(F)
  6 │     }
  7 │     
  8 │ ╭─▶ def Foo(?).foo: Foo(F) {
    · ││        ─
  9 │ │       Bar => ?,
 10 │ │       Baz => ?
 11 │ ├─▶ }
    · ╰──── While elaborating
    ╰────
```

However, it would be better not to use the span of the whole definition, but the span of the case `Bar => ?`.
Similarly for top-level codefinitions and local (co)matches.

Reply:
I.e. instead of passing the span from the top-level definition to the case in `check_ws`, use the span of the case.

Reply:
![image](https://github.com/user-attachments/assets/29e56668-1e86-4d06-a7db-b6e8251276ad)

ah yea that appears to be much better :)

Reply:
![image](https://github.com/user-attachments/assets/faa0f3b4-65da-4a4c-b2f7-8c7c559cb9d9)
also works for comatches :)